### PR TITLE
[#6090] add teaser block WITH MIGRATION!

### PR DESCRIPTION
**Pull request by phillimorland**

### CODE REVIEW

File: apps/academy/blocks.py [hunk 1]
```diff
@@ -1,30 +1,17 @@
...
-        help_text='Add an academy page')
-
-
-class ChallengeExternalLinkBlock(StructBlock):
-    challenge_step_external_links = PageChooserBlock(
-        target_model='academy.AcademyExternalLink',
-        required=False,
-        help_text='Add a non academy page')
-
-
```
Comment (phillimorland) [inline apps/academy/blocks.py:19]: was un-used

File: apps/academy/blocks.py [hunk 2]
```diff
@@ -56,3 +48,45 @@ class Meta:
...
+    )
+    anchor_link = CharBlock(
+        required=False, max_length=28, label='Anchor Link',
+        help_text="Anchor link should be all one word."
+
+    )
+
+
+class AcademyCallToActionBlock(StructBlock):
+    columns_count = ChoiceBlock(choices=[
```
Comment (fuzzylogic2000) [inline apps/academy/blocks.py:78]: We don't need the columns count, right?

**Review by fuzzylogic2000 — CHANGES_REQUESTED** [review]
Looks great! But I think we can remove the column count?

File: apps/academy/blocks.py [hunk 3]
```diff
@@ -56,3 +48,45 @@ class Meta:
...
+    columns = ListBlock(
+        CallToActionBlock(label='List and Image'),
+        max_num=2
+    )
+
+    class Meta:
+        template = 'academy/blocks/block_col_cta.html'
+        icon = 'plus-inverse'
+        help_text = '1 or 2 column teaser with black '
+        'background and white writing'
```
[the remark this answers is not in the document]
Comment (phillimorland) [reply apps/academy/blocks.py:92]: not in the meta class just in the block, at least that's how we do it in other files but you're python person so we can change it if you prefer? got it from here https://github.com/liqd/liqd-site/blob/1e9fddaac3c773f49e05526e155cc24aec9d890e/apps/core/blocks.py#L15

File: apps/academy/templates/academy/blocks/block_col_cta.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{% load wagtailimages_tags %}
+<section class="block block__col-cta--dark">
+    <div class="container">
+        <div class="row justify-content-center">
+            {% for column in value.columns %}
+            {% if forloop.counter == 2 %}
```
[the remark this answers is not in the document]
Comment (phillimorland) [reply apps/academy/templates/academy/blocks/block_col_cta.html:6]: cool as column count was not working :)

File: apps/academy/blocks.py [hunk 5]
```diff
@@ -56,3 +46,39 @@ class Meta:
...
+    body_text = TextBlock(required=True, length=120)
+    cta_link = StructBlock([
+        ('internal_link', PageChooserBlock(required=False,
+                                           help_text="The external link "
+                                           "overwrites the link to a local "
+                                           "page. Please only add 1 link.")),
+        ('external_link', URLBlock(required=False)),
+    ])
+    link_text = CharBlock(
+        required=False, max_length=28, label='Link Text'
```
Comment (fuzzylogic2000) [inline apps/academy/blocks.py:66]: I am so sorry, that I only catched that this time and not with the first time of redoing the migrations, but the link_text needs to be required.

Comment (fuzzylogic2000) [reply apps/academy/blocks.py:66]: Or do we also want to be able to use it without link? I guess, yes! :sweat_smile:

**Review by fuzzylogic2000 — APPROVED** [review]
Nice!

**Pull request merged**